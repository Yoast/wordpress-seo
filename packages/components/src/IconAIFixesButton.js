import React from "react";
import PropTypes from "prop-types";

/* Yoast dependencies */
import { colors } from "@yoast/style-guide";

/* Internal dependencies */
import AIFixesButton from "./AIFixesButton";

/**
 * Returns the IconAIFixesButton component.
 *
 * @param {Object} props Component props.
 *
 * @returns {JSX.Element} IconAIFixesButton component.
 */
const IconAIFixesButton = function( props ) {
	return (
		<AIFixesButton
			disabled={ props.disabled }
			type="button"
			onClick={ props.onClick }
			pressed={ props.pressed }
			onPointerEnter={ props.onPointerEnter }
			onPointerLeave={ props.onPointerLeave }
			unpressedBoxShadowColor={ props.unpressedBoxShadowColor }
			pressedBoxShadowColor={ props.pressedBoxShadowColor }
			pressedBackground={ props.pressedBackground }
			unpressedBackground={ props.unpressedBackground }
			id={ props.id }
			aria-label={ props.ariaLabel }
			aria-pressed={ props.pressed }
			pressedIconColor={ props.pressedIconColor }
			className={ props.className }
		>
			{ props.children }
		</AIFixesButton>
	);
};

IconAIFixesButton.propTypes = {
	disabled: PropTypes.bool,
	children: PropTypes.node,
	id: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string,
	onClick: PropTypes.func,
	onPointerEnter: PropTypes.func,
	onPointerLeave: PropTypes.func,
	unpressedBoxShadowColor: PropTypes.string,
	pressedBoxShadowColor: PropTypes.string,
	pressedBackground: PropTypes.string,
	unpressedBackground: PropTypes.string,
	pressedIconColor: PropTypes.string,
	pressed: PropTypes.bool.isRequired,
	className: PropTypes.string,
};

IconAIFixesButton.defaultProps = {
	disabled: false,
	unpressedBoxShadowColor: colors.$color_button_border,
	pressedBoxShadowColor: colors.$color_purple,
	pressedBackground: "linear- gradient(to bottom right, #A61E69, #3B82F6)",
	unpressedBackground: "linear-gradient(to bottom right, #FAF3F7, #EFF6FF)",
	pressedIconColor: colors.$color_white,
	onClick: () => {},
	onPointerEnter: () => {},
	onPointerLeave: () => {},
};

export default IconAIFixesButton;
