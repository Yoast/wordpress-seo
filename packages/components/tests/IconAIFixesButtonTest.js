import React from "react";
import renderer from "react-test-renderer";

import IconAIFixesButton from "../src/IconAIFixesButton";

test( "the unpressed IconAIFixesButton matches the snapshot", () => {
	const component = renderer.create(
		<IconAIFixesButton
			name="group1"
			id="keyphraseDensityAIFixes"
			ariaLabel="Fix with AI"
			pressed={ false }
			disabled={ true }
			onClick={ () => {} }
			className={ "yoast-tooltip yoast-tooltip-w" }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the pressed IconAIFixesButton matches the snapshot", () => {
	const component = renderer.create(
		<IconAIFixesButton
			name="group1"
			id="keyphraseInIntroductionAIFixes"
			ariaLabel="Fix with AI"
			pressed={ true }
			disabled={ false }
			onClick={ () => {} }
			className={ "yoast-tooltip yoast-tooltip-w" }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

