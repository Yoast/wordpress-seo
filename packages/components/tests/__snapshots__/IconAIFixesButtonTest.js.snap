// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the pressed IconAIFixesButton matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  min-width: 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid #ccc;
  background-color: linear- gradient(to bottom right,#A61E69,#3B82F6);
  box-shadow: inset 0 2px 0 rgba( 93,35,122,0.7 );
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: 24px;
}

.c0:disabled {
  background-color: linear-gradient(to bottom right,#FAF3F7,#EFF6FF);
  box-shadow: none;
  border: none;
  cursor: default;
}

.c1 {
  position: relative;
  border: none;
  background-image: linear-gradient(to bottom right,#A61E69,#6366F1) !important;
  box-shadow: inset 0 -2px 0 #B94986;
}

.c1:hover {
  background-image: linear-gradient(to bottom right,#A61E69,#6366F1) !important;
}

<button
  aria-label="Fix with AI"
  aria-pressed={true}
  className="c0 c1 yoast-tooltip yoast-tooltip-w"
  disabled={false}
  id="keyphraseInIntroductionAIFixes"
  onClick={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  type="button"
/>
`;

exports[`the unpressed IconAIFixesButton matches the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  min-width: 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid #ccc;
  background-color: linear-gradient(to bottom right,#FAF3F7,#EFF6FF);
  box-shadow: 0 1px 0 rgba( 204,204,204,0.7 );
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: 24px;
}

.c0:disabled {
  background-color: linear-gradient(to bottom right,#FAF3F7,#EFF6FF);
  box-shadow: none;
  border: none;
  cursor: default;
}

.c1 {
  position: relative;
  border: 1px solid #A5B4FC;
  background-image: linear-gradient(to bottom right,#FAF3F7,#EEF2FF) !important;
  box-shadow: 0 1px 0 rgba( 204,204,204,0.7 );
}

.c1:hover {
  background-image: linear-gradient(to bottom right,#F3E5ED,#E0E7FF) !important;
}

<button
  aria-label="Fix with AI"
  aria-pressed={false}
  className="c0 c1 yoast-tooltip yoast-tooltip-w"
  disabled={true}
  id="keyphraseDensityAIFixes"
  onClick={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  type="button"
/>
`;
